ethereum-consensus/src/capella/block_processing.rs

```diff
@@ -39,10 +42,57 @@ pub fn process_bls_to_execution_change<
         BYTES_PER_LOGS_BLOOM,
         MAX_EXTRA_DATA_BYTES,
     >,
-    _signed_address_change: &mut SignedBlsToExecutionChange,
-    _context: &Context,
+    signed_address_change: &mut SignedBlsToExecutionChange,
+    context: &Context,
 ) -> Result<()> {
-    unimplemented!()
+    let address_change = &mut signed_address_change.message;
+    let signature = &signed_address_change.signature;
+
+    if address_change.validator_index >= state.validators.len() {
+        return Err(invalid_operation_error(InvalidOperation::BlsToExecutionChange(
+            InvalidBlsToExecutionChange::ValidatorIndexOutOfBounds(address_change.validator_index),
+        )))
+    }
+
+    let validator = &mut state.validators[address_change.validator_index];
+    let withdrawal_credentials = &mut validator.withdrawal_credentials;
+    if withdrawal_credentials[0] != BLS_WITHDRAWAL_PREFIX {
+        return Err(invalid_operation_error(InvalidOperation::BlsToExecutionChange(
+            InvalidBlsToExecutionChange::WithdrawalCredentialsPrefix(
+                state.validators[address_change.validator_index].withdrawal_credentials[0],
+            ),
+        )))
+    }
+
+    let public_key = &address_change.from_bls_public_key;
+
+    if withdrawal_credentials[1..] != hash(public_key.as_slice())[1..] {
+        return Err(invalid_operation_error(InvalidOperation::BlsToExecutionChange(
+            InvalidBlsToExecutionChange::PublicKeyMismatch(public_key.clone()),
+        )))
+    }
+
+    let domain = compute_domain(
+        DomainType::BlsToExecutionChange,
+        None,
+        Some(state.genesis_validators_root),
+        context,
+    )?;
+
+    let signing_root = compute_signing_root(address_change, domain)?;
+    verify_signature(&address_change.from_bls_public_key, signing_root.as_ref(), signature)
+        .map_err(|_| {
+            invalid_operation_error(InvalidOperation::BlsToExecutionChange(
+                InvalidBlsToExecutionChange::InvalidSignature(signature.clone()),
+            ))
+        })?;
+
+    validator.withdrawal_credentials[0] = ETH1_ADDRESS_WITHDRAWAL_PREFIX;
+    validator.withdrawal_credentials[1..12].fill(0);
+    validator.withdrawal_credentials[12..]
+        .copy_from_slice(address_change.to_execution_address.as_ref());
```

Comment:
if we are going to take `&mut` on the `withdrawal_credentials` directly, let's just use that

```suggestion
    withdrawal_credentials[0] = ETH1_ADDRESS_WITHDRAWAL_PREFIX;
    withdrawal_credentials[1..12].fill(0);
    withdrawal_credentials[12..]
        .copy_from_slice(address_change.to_execution_address.as_ref());
```